test/JDBC/input/BABEL_4877.mix

```diff
@@ -0,0 +1,178 @@
+-- tsql
+CREATE SCHEMA babel_4877_s1
+GO
+
+CREATE FUNCTION babel_4877_s1.babel_4877_f1() RETURNS INT
+AS
+BEGIN
+    DECLARE @babel_4877_tablevar TABLE (id INT)
+    INSERT INTO @babel_4877_tablevar SELECT id FROM babel_4877_t;
+    RETURN (SELECT id FROM @babel_4877_tablevar)
+END
+GO
+
+CREATE PROC babel_4877_s1.babel_4877_p
+AS
+    SELECT babel_4877_s1.babel_4877_f1()
+GO
+
+CREATE TABLE dbo.babel_4877_t (id INT)
+GO
+INSERT INTO dbo.babel_4877_t VALUES (10)
+GO
+
+-- func in s1 should see dbo table
+EXEC babel_4877_s1.babel_4877_p
+GO
+
+CREATE TABLE babel_4877_s1.babel_4877_t (id INT)
+GO
+INSERT INTO babel_4877_s1.babel_4877_t VALUES (20)
+GO
+
+-- Now func in s1 should select unspecified objects from s1 first then dbo
+EXEC babel_4877_s1.babel_4877_p
+GO
+
+SELECT babel_4877_s1.babel_4877_f1()
+GO
+
+DROP TABLE babel_4877_s1.babel_4877_t, dbo.babel_4877_t
+GO
+DROP PROC babel_4877_s1.babel_4877_p
+GO
+DROP FUNCTION babel_4877_s1.babel_4877_f1
+GO
+DROP SCHEMA babel_4877_s1
+GO
+
+
+-- psql
+CREATE PROCEDURE master_dbo.babel_4877_reset_search_path() AS
+$$ BEGIN
+RESET search_path;
+END $$
+LANGUAGE plpgsql;
+GO
+
+-- tsql
+CREATE SCHEMA babel_4877_s1
+GO
+CREATE FUNCTION babel_4877_s1.babel_4877_f1() RETURNS VARCHAR(100)
+AS
+BEGIN
+    RETURN (SELECT set_config('search_path','master_babel_4877_s1, sys, pg_catalog', false));
+END
+GO
+CREATE TABLE babel_4877_t1 (id INT);
+GO
+INSERT INTO babel_4877_t1 VALUES (generate_series(1,10));
+GO
+
+-- user set commands should persists even after the function call ends
+CREATE PROC babel_4877_s1.babel_4877_p1 AS
+    SELECT 'before search path ====> ', current_setting('search_path');
+    SELECT babel_4877_s1.babel_4877_f1() FROM babel_4877_t1;
+    SELECT 'after search path ====> ', current_setting('search_path');
+GO
+
+EXEC babel_4877_s1.babel_4877_p1
+GO
+
+DROP FUNCTION babel_4877_s1.babel_4877_f1
+GO
+
+EXEC dbo.babel_4877_reset_search_path
+GO
+
+-- now search path = reset val 
+
+CREATE FUNCTION babel_4877_s1.babel_4877_f1() RETURNS VARCHAR(100)
```

Comment:
can you try this scenario?
begin tran();
check search path;
call f1(); -> throw an error which is stmt terminating (divide by zero or similar)
check search path;
commit;

Reply:
added test cases